### ISSUE TITLE
Using MapDynamicControllerRoute("{**path}") throws NullReferenceException:

### ISSUE THREAD

**Issue by alienwareone** [microsoft/aspnet-api-versioning#658]
I've attached a test project:
https://github.com/alienwareone/DynamicEndpointRoutingApiVersioningBug

```
System.NullReferenceException: Object reference not set to an instance of an object.
   at Microsoft.AspNetCore.Mvc.Routing.ApiVersionMatcherPolicy.TrySelectApiVersion(HttpContext httpContext, CandidateSet candidates)
   at Microsoft.AspNetCore.Mvc.Routing.ApiVersionMatcherPolicy.ApplyAsync(HttpContext httpContext, CandidateSet candidates)
   at Microsoft.AspNetCore.Routing.Matching.DfaMatcher.SelectEndpointWithPoliciesAsync(HttpContext httpContext, IEndpointSelectorPolicy[] policies, CandidateSet candidateSet)
   at Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatch|8_1(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task matchTask)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
```

**Comment by commonsensesoftware** [comment]
This is a duplicate of #600. I have a large number of fixes in the works. The will be included in the next patch version. The fix is pretty straight forward - fortunately. Thanks.

**Comment by commonsensesoftware** [comment]
This issue has now been fixed in [5.0.0-RC.1](https://www.nuget.org/packages/Microsoft.AspNetCore.OData.Versioning/5.0.0-rc.1). The official release will likely occur after 2 weeks of _burn-in_. If that doesn't fix the issue, feel free to reopen this issue or open a new one. Thanks.
